system/atomic_image.py

```diff
@@ -0,0 +1,109 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION='''
+---
+module: atomic_image
+short_description: Manage the container images on the atomic host platform
+description:
+    - Manage the container images on the atomic host platform
+    - Allows to execute the commands on the container images
+    - Commands ```install``` and ```run``` will download the container image if not present already
+version_added: "2.2"
+author: "Saravanan KR @krsacme"
+notes:
+    - Host should be an atomic platform (verified by existence of '/run/ostree-booted' file)
+requirements:
+  - atomic
+options:
+    name:
+        description:
+          - Name of the container image
+        required: True
+        default: null
+    command:
+        description:
+          - The command to run on the container
+        required: True
+        choices: ["install", "run", "stop", "uninstall"]
+        default: null
+'''
+
+EXAMPLES = '''
+
+# Execute the install command on rhel-tools container image (atomic install rhel7/rhel-tools)
+- atomic_image: name=rhel7/rhel-tools command=install
+
+# Execute the run command on rsyslog container image (atomic run rhel7/rsyslog)
+- atomic_image: name=rhel7/rsyslog command=run
+
+'''
+
+RETURN = '''
+msg:
+    description: The command standard output
+    returned: always
+    type: string
+    sample: [u'Using default tag: latest ...']
+'''
+
+import os
+import sys
+
+def core(module):
+    image = module.params['name']
+    command = module.params['command']
+    args = ['atomic']
+    args.append(command)
+    args.append(image)
+
+    out = {}
+    err = {}
+    rc = 0
+
+    rc, out, err = module.run_command(args, check_rc=False)
+
+    if rc != 0: # something went wrong emit the msg
+        module.fail_json(rc=rc, msg=err)
+    elif command == "run" and "Container is running" in out:
+        module.exit_json(result=out)
+    else:
+        module.exit_json(msg=out, changed=True)
+
+
+def main():
+    module = AnsibleModule(
+                argument_spec = dict(
+                    name = dict(default=None, required=True),
+                    command = dict(default=None, choices=['install', 'run', 'stop', 'uninstall'], required=True),
+                ),
+            )
+
+    # Verify that the platform is atomic host
+    if not os.path.exists("/run/ostree-booted"):
+      module.fail_json(msg="Module atomic_image is applicable for Atomic Host Platforms only")
```

Comment:
OK. I will add a check to verify the atomic command existence as the only pre-condition for this module.